examples/ed25519/src/ed25519.rs

```diff
@@ -0,0 +1,763 @@
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0x7fu8 
+    ]
+));
+
+#[rustfmt::skip]
+const CONSTANT_L: SerializedScalar = SerializedScalar(secret_array!(
+    U8, 
+    [
+        0xedu8, 0xd3u8, 0xf5u8, 0x5cu8, 0x1au8, 0x63u8, 0x12u8, 0x58u8,
+        0xd6u8, 0x9cu8, 0xf7u8, 0xa2u8, 0xdeu8, 0xf9u8, 0xdeu8, 0x14u8,
+        0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x00u8, 
+        0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x00u8, 0x10u8 
+    ]
+));
+
+#[rustfmt::skip]
+const CONSTANT_P3_8: SerializedScalar = SerializedScalar(secret_array!(
+    U8, 
+    [
+        0xfeu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0x0fu8 
+    ]
+));
+
+#[rustfmt::skip]
+const CONSTANT_P1_4: SerializedScalar = SerializedScalar(secret_array!(
+    U8, 
+    [
+        0xfbu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 
+        0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0xffu8, 0x1fu8 
+    ]
+));
+
+#[rustfmt::skip]
+const CONSTANT_D: SerializedScalar = SerializedScalar(secret_array!(
+    U8, 
+    [
+        0xa3u8, 0x78u8, 0x59u8, 0x13u8, 0xcau8, 0x4du8, 0xebu8, 0x75u8, 
+        0xabu8, 0xd8u8, 0x41u8, 0x41u8, 0x4du8, 0x0au8, 0x70u8, 0x00u8, 
+        0x98u8, 0xe8u8, 0x79u8, 0x77u8, 0x79u8, 0x40u8, 0xc7u8, 0x8cu8, 
+        0x73u8, 0xfeu8, 0x6fu8, 0x2bu8, 0xeeu8, 0x6cu8, 0x03u8, 0x52u8 
+    ]
+));
+
+fn is_negative(x: Ed25519FieldElement) -> U8 {
+    if x.bit(0) {
+        U8(1u8)
+    } else {
+        U8(0u8)
+    }
+}
+
+fn compress(p: EdPoint) -> CompressedEdPoint {
+    let (x, y, z, _) = p;
+    let z_inv = z.inv();
+    let x = x * z_inv;
+    let y = y * z_inv;
+    let mut s: ByteSeq = y.to_byte_seq_le();
+    s[31] = s[31] ^ (is_negative(x) << 7); // setting top bit
+    CompressedEdPoint::from_slice(&s, 0, 32)
+}
+
+fn sqrt(a: Ed25519FieldElement) -> Option<Ed25519FieldElement> {
+    let p3_8 = Ed25519FieldElement::from_byte_seq_le(CONSTANT_P3_8);
+    let p1_4 = Ed25519FieldElement::from_byte_seq_le(CONSTANT_P1_4);
+
+    let x_c = a.pow_self(p3_8);
+    let mut result: Option<Ed25519FieldElement> = Option::<Ed25519FieldElement>::None;
+    if x_c * x_c == a {
+        result = Some(x_c);
+    };
+    if x_c * x_c == Ed25519FieldElement::ZERO() - a {
+        let x = Ed25519FieldElement::TWO().pow_self(p1_4) * x_c;
+        result = Some(x);
+    }
+    result
+}
+
+fn check_canonical_point(x: CompressedEdPoint) -> bool {
```

Comment:
With this signature you can skip the next line.
```suggestion
fn check_canonical_point(mut x: CompressedEdPoint) -> bool {
```